### ISSUE TITLE
`mirror_account_chart` fails if account root category has no sub-nodes

### ISSUE THREAD

**Issue by lasuk** [macxred/cashctrl_ledger#41]
**Issue**

`mirror_account_chart(..., delete=True)` fails when an account root category such as `/Balance` has no sub-nodes.
```
>>> cc_ledger.mirror_account_chart(accounts, delete=True)
requests.exceptions.RequestException: API call failed. A system-category cannot be deleted.
```

**Reason**

1. Account category root nodes are immutable in CashCtrl.
2. We have implemented a trick to exclude root categories from being deleted, but this trick fails to exclude root categories without a nested sub-category. See [ledger.py#L140](https://github.com/macxred/cashctrl_ledger/blob/7cc2d6fab7/cashctrl_ledger/ledger.py#L140)

**Proposed Solution**

Explicitely ignore mutations of root account categories in the low-level function that handles categories.

1. Add an option `ignore_account_root_nodes: bool = False` to `CashCtrlClient.update_categories()`. When set, we silently ignore additions, updates and deletions of root account categories. 
2. Modify `mirror_account_chart()` to activate this option. Remove trick on  [ledger.py#L140](https://github.com/macxred/cashctrl_ledger/blob/7cc2d6fab7/cashctrl_ledger/ledger.py#L140) that ignores root node most of the time but fails to work in some circumstance.

**Comment by lasuk** [comment]
A preliminary implementation of step 1 is available in macxred/cashctrl_api#32. 
- The current stat covers updating and deletion, but not the addition, of account root nodes. 
- The new functionality is not yet covered by tests.

**Comment by AlexTheWizardL** [comment]
Implemented in #40
